### ISSUE TITLE
Incorporate transmission & adjacent markets as a concept

### ISSUE THREAD

**Issue by toddmedema** [toddmedema/electrify#61]
1. Lets say for simplicity that you as the power company have a monopoly on your area, so that we can abstract away the challenges inside your zone
2. Then, you have zones around you (it'd be awesome if each scenario grabbed historically and locally accurate transmission info), perhaps defined as 1-6 connected regions, with a set amount of transmission capacity
3. Then, your levers as a player are: (perhaps this is a new tab)

### Buy and sell electricity with each region

Each region should be willing to either buy or sell a certain quantity of electricity at a certain price (or, more accurately, a curve, e.g. the more you buy/sell, the less valuable it becomes). 

This will also be constrained by transmission infrastructure, which is not a Boolean / hard limit - the hotter the temperature, the less current lines can support due to sagging (TODO research what's the drop, ideally simplified to a -x% per degree C above y temperature)

TODO: BIG UI question: How to do this? Buying/selling electricity is an ongoing concept, so it's not a page you can go to and click "sell" when you have spare capacity. It's almost like you need a rules engine, sell to the highest buyer when you have spare capacity, buy from the lowest seller when you have a capacity gap... But then players will definitely want tools to debug (for example, seeing how much of their current supply/demand/income/cost is going to/from trading, forecasts on trading pricing and availability, indications when your trading is constrained by transmission capacity)

TODO: research simplified factors of how we can calculate adjacent region's quantity and price 

### Upgrade transmission infrastructure

Like generation facilities, should have a sliding scale with how much capacity you'd like to add, and the resulting cost + time to build.

UI can probably be similar to existing list of generators; perhaps there's a high level list of all regions, then when you click one region, you see the specific transmission lines (having the concept of specific lines would then allow us, for example, to have events where a specific line goes out of commission)

TODO: research needed on factors that influence the cost and timeframes for transmission construction

**Comment by toddmedema** [comment]
## Proposed implementation plan

I recommend implementing transmission as two connected concepts:

- **Interties** are infrastructure managed alongside plants and storage.
- **Power exchange** is an Insights layer showing trades, constraints, prices, and forecasts.

I would not add a fourth global navigation destination. On phones it would crowd the footer, and
on desktop it would imply another permanent pane. A local `Plants | Interties` switch inside
Facilities fits the existing information architecture.

**Selected icon:** use `public/images/transmission-option-2.svg` (the bidirectional exchange icon with two regional nodes) throughout the Facilities intertie and Power exchange interfaces. Preserve its 40/20/10 stroke hierarchy and clean, symmetric geometry.

The values in these mocks are illustrative; final prices, costs, and ratings should come from
calibrated game data.

### Desktop operations

![Desktop mock showing Interties in Facilities and Power exchange in Insights](https://github.com/user-attachments/assets/ace8fae2-8b6d-4597-ad2b-7d69494d518f)

### Phone operations

![Phone mock showing Interties without changing the three-item bottom navigation](https://github.com/user-attachments/assets/af9debc4-8171-4765-aa37-9f4a0d7267df)

### Build flow

![Build transmission mock with region, capacity, upgrade, and new-corridor choices](https://github.com/user-attachments/assets/4c397878-bbe0-4e59-983a-e738ce04cf1b)

### 1. Introduce the domain model

Add canonical types in `src/Types.tsx`:

- `AdjacentMarketDefinition`: region identity, reference prices, import/export quantity curves,
  and seasonality.
- `TransmissionCorridorDefinition`: endpoints, length, voltage/class, base capacity, and
  construction options.
- `TransmissionLineOperatingType`: current capacity, construction progress, financing, outages,
  and derates.
- `TradingPolicyType`: import cap, export floor, reserve target, and blackout-protection setting.
- Per-tick regional exchange data: price, available quantity, effective line rating, imports,
  exports, and curtailed trade.
- Monthly totals for imported/exported energy, import cost, export revenue, and congestion.

Add optional adjacent-market definitions to `ScenarioType` and mutable lines/policy to `GameType`.
Keep islands such as Paradise and Hurricane Season intentionally disconnected.

### 2. Build deterministic market data

Create `src/data/AdjacentMarkets.tsx` containing authored, offline data. Runtime API calls would
make saves and replays non-deterministic.

- Use [EIA-930 neighboring balancing-authority interchange data](https://www.eia.gov/opendata/browser/electricity/rto/interchange-data)
  to calibrate US topology and typical transfer volumes.
- Generate deterministic price/quantity blocks from regional baselines, demand shape, fuel
  prices, temperature, time of day, season, and the game seed.
- Start with one well-researched California pilot, then cover other connected scenarios. Treat
  European regions as a separate data pass instead of inventing US-style balancing authorities.

### 3. Model transmission ratings

Add pure helpers in `src/helpers/Transmission.tsx`.

Avoid a universal “−x% per degree” rule. Generate small lookup tables for representative conductor
classes under documented assumptions:

- Ambient temperature.
- Day/night solar heating.
- Conservative assumed wind.
- Normal and emergency rating floors.

FERC distinguishes ambient-adjusted ratings from dynamic ratings; dynamic ratings additionally use
wind, cloud cover, precipitation, tension, and sag. That supports an ambient-plus-solar first
version, with full dynamic ratings deferred. See the
[FERC Order 881 overview](https://www.ferc.gov/news-events/news/staff-presentation-final-order-regarding-managing-transmission-line-ratings).

### 4. Add market clearing to the simulation

Integrate a pure `clearAdjacentMarkets` step into the forecast/dispatch path in
`src/reducers/Game.tsx`.

For each tick:

1. Compute each line's temperature-adjusted transfer limit.
2. Preserve the existing facility dispatch order.
3. Buy from the cheapest eligible regions when local capacity cannot meet demand plus the
   configured reserve, subject to quantity, line limit, and import price cap.
4. Offer surplus generation to the highest-paying regions after local demand and reserve are
   protected.
5. Dispatch extra export generation only when the regional bid exceeds incremental fuel, O&M,
   carbon, and start costs.
6. Prevent simultaneous importing and exporting over the same corridor and prohibit circular
   arbitrage.
7. Record import expenses and export revenue separately before updating cash and net worth.

FERC's simplified LMP explanation—energy cost plus losses and congestion—is the right conceptual
basis for the price curve without attempting a full nodal power-flow model. See
[FERC's energy-market overview](https://www.ferc.gov/understanding-energy-markets).

### 5. Add construction and financing

Reuse the capacity-slider pattern from `BuildGenerators.tsx` and offer two corridor classes:

- Upgrade/reconduct an existing right-of-way: faster, cheaper, and capped expansion.
- Build a new high-voltage corridor: slower and more expensive, with greater capacity.

Author length and terrain multipliers per corridor, then apply scenario difficulty. DOE gives
roughly $2 million/mile as one example for a 230-kV Iowa line while emphasizing strong location
dependence, and notes that new transmission commonly takes 5–10 years to plan and build. These
should anchor—not dictate—the tuned game catalog. See DOE's
[advanced-conductor report](https://www.energy.gov/sites/default/files/2024-08/Advanced%20Conductor%20Report%20December%202023.pdf)
and [development guide](https://www.energy.gov/cmei/systems/windexchange/land-based-economic-development-guide).

### 6. Implement the UI

In `Facilities.tsx`:

- Add `Plants | Interties`.
- Show each adjacent region's live direction, flow, price, rating, utilization, and construction
  state.
- Expand a region to reveal individual corridors and the reason for any constraint.
- Show `Upgrade` only while Interties is selected.

In `Insights.tsx`:

- Add a `Power exchange` layer to the Grid preset.
- Chart imports, exports, and temperature-adjusted limits.
- Add trading policy to Planning controls.
- Add finance-detail rows for import cost, export revenue, and net trading income.
- Produce actionable congestion and line-outage events in the existing event log.

### 7. Compatibility and rollout

- Let old saves without transmission state continue as legacy runs; initialize the feature only
  for new games. This should avoid invalidating saves.
- Bump `REPLAY_VERSION`, because changed scenario initialization and dispatch will alter recorded
  outcomes.
- Add a focused transmission tutorial after the existing forecast tutorial.
- Ship the California pilot first, balance it, then add regions to other scenarios.
- Add line outages and targeted world-event effects only after the core market is stable.

### Acceptance criteria

- Imports prevent a blackout only when market supply, policy, and effective line capacity permit
  it.
- Exports never reduce local service below the configured reserve target.
- Trade respects regional quantity curves and every active corridor limit.
- Hot conditions visibly reduce applicable line ratings and forecasts.
- Import costs and export revenue reconcile exactly with cash movement.
- Forecasts never mutate live facilities, lines, or lifetime totals.
- The same seed and replay produce identical trades and outcomes.
- Old saves load without gaining unexpected interties mid-run.
- Facilities and Insights remain usable without horizontal overflow at 320px.
- Add helper, reducer, component, save/replay, invariant, simulation, and Playwright coverage.
- Finish with `npm run check`, `npm run sim -- --all`, and the relevant responsive E2E suites.
